src/java.base/share/classes/java/util/concurrent/ForkJoinWorkerThread.java

```diff
@@ -70,10 +72,13 @@ public class ForkJoinWorkerThread extends Thread {
      * Full nonpublic constructor.
      */
     ForkJoinWorkerThread(ThreadGroup group, ForkJoinPool pool,
-                         boolean useSystemClassLoader, boolean isInnocuous) {
-        super(group, null, pool.nextWorkerThreadName(), 0L);
+                         boolean useSystemClassLoader,
+                         boolean clearThreadLocals) {
```

Comment:
It's tempting to toggle the sense of last boolean argument to be `preserveThreadLocals` for consistency (given the multiple toggles as the value is propagated from the newly added protected constructor), but which means toggling the value at the use sites that you may not wish to change.

Reply:
Thanks. I changed param name to clearThreadLocals, which I think helps. I do need (only) one negation somewhere along these paths because internally clearing requires a set config bit.